pkg/syncer/volume_health_reconciler.go

```diff
@@ -91,16 +149,74 @@ func NewVolumeHealthReconciler(
 		svcPVCSynced:        svcPVCInformer.Informer().HasSynced,
 		svcClaimQueue:       svcClaimQueue,
 		supervisorNamespace: supervisorNamespace,
+		volumeHandleToPVs: &volumeHandleToPVs{
+			RWMutex: &sync.RWMutex{},
+			items:   make(map[string][]string),
+		},
 	}
 
 	svcPVCInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
 		AddFunc:    rc.svcAddPVC,
 		UpdateFunc: rc.svcUpdatePVC,
```

Comment:
I remember discussing this but dont recollect the conclusion - any reason why we are not handling SV PVC delete events?

Reply:
IIRC, we decided to address that in a follow up PR

Reply:
Ok, filed this issue - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/433